cylc/flow/graph_parser.py

```diff
@@ -335,191 +454,409 @@ def _report_invalid_lines(cls, lines):
+            for outp in [TASK_OUTPUT_SUCCEEDED, TASK_OUTPUT_FAILED]:
+                self._set_output_opt(
+                    name, outp, optional, suicide, fam_member)
+
+        try:
+            prev_optional, prev_default, prev_fixed = (
+                self.task_output_opt[(name, output)])
+        except KeyError:
+            # Not already set; set it. Fix it if not fam_member.
+            self.task_output_opt[(name, output)] = (
+                optional, optional, not fam_member)
+        else:
+            # Already set; check consistency with previous.
+            if prev_fixed:
+                # optionality fixed already
+                if fam_member:
+                    pass
+                else:
+                    if optional != prev_optional:
+                        raise GraphParseError(
+                            f"Output {name}:{output} can't be both required"
+                            " and optional")
+            else:
+                # optionality not fixed yet (only family default)
+                if fam_member:
+                    # family defaults must be consistent
+                    if optional != prev_default:
+                        raise GraphParseError(
+                            f"Output {name}:{output} can't default to both"
+                            " optional and required (via family trigger"
+                            " defaults)")
+                else:
+                    # fix the optionality now
+                    self.task_output_opt[(name, output)] = (
+                        optional, prev_default, True)
+
+        # Check opposite output where appropriate.
+        for opposites in [
+            (TASK_OUTPUT_SUCCEEDED, TASK_OUTPUT_FAILED),
+            (TASK_OUTPUT_SUBMITTED, TASK_OUTPUT_SUBMIT_FAILED)
+        ]:
+            if output not in opposites:
+                continue
+            succeed, fail = opposites
+            opposite = fail if output == succeed else succeed
+            try:
+                opp_optional, opp_default, opp_fixed = (
+                    self.task_output_opt[(name, opposite)]
+                )
+            except KeyError:
+                # opposite not set, no need to check
+                continue
+            else:
+                # opposite already set; check consistency
+                optional, default, oset = (
+                    self.task_output_opt[(name, output)]
+                )
+                msg = (f"Opposite outputs {name}:{output} and {name}:"
+                       f"{opposite} must both be optional if both are used")
+                if fam_member or not opp_fixed:
+                    if not optional or not opp_default:
+                        raise GraphParseError(
+                            msg + " (via family trigger defaults)")
+                    elif not optional or not opp_optional:
+                        raise GraphParseError(
+                            msg + " (via family trigger)")
```

Comment:
Would be nice to cover this with a test

Reply:
Huh, it should be covered already.
```
$ git grep 'via family trigger' tests
tests/unit/test_graph_parser.py:            ("must both be optional if both are used (via family trigger"
tests/unit/test_graph_parser.py:            ("can't default to both optional and required (via family trigger"
```

Reply:
Must have been a coverage glitch, the Codecov warning is gone now